### PR TITLE
CI fragmentation

### PR DESCRIPTION
**Pull request by gsvgit**
Part of #100 .
- Run USVM tests only for changes in respective submodule.
- Do not run usvm tests  in build-and-launch. Just build USVM and use it for training.

### CODE REVIEW

File: .github/workflows/build_and_test_usvm.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+name: Build and test USVM submodule
+
+on: 
+  push:
+    paths: 
+      - 'GameServers/usvm/'
+  pull_request:
+    paths: 
+      - 'GameServers/usvm/'
```
Comment (emnigma) [inline .github/workflows/build_and_test_usvm.yaml:9]: Doesn't seem to work: https://github.com/PySymGym/PySymGym/pull/102

Comment (emnigma) [reply .github/workflows/build_and_test_usvm.yaml:9]: if it indeed does not work, we may try more direct approach:
```yaml
      - name: Check for submodule changes
        run: |
          git fetch --unshallow
          old_sha=$(git rev-parse HEAD^1:GameServers/usvm)
          new_sha=$(git rev-parse HEAD:GameServers/usvm)
          if [ "$old_sha" != "$new_sha" ]; then
            ./gradlew build
          else
            ./gradlew assemble
          fi
```

Comment (gsvgit) [reply .github/workflows/build_and_test_usvm.yaml]: @emnigma Looks fixed: #102

Comment (emnigma) [reply .github/workflows/build_and_test_usvm.yaml]: Successful run actually **has** trailing slashes in gh-actions workflow file

image

Im almost certain that check is triggered if usvm is updated **and** some changes were introduced in main repo: 

image

Comment (emnigma) [reply .github/workflows/build_and_test_usvm.yaml]: Judging by ci runs, this does work (https://github.com/PySymGym/PySymGym/pull/103)

> if it indeed does not work, we may try more direct approach:
> 
> ```yaml
>       - name: Check for submodule changes
>         run: |
>           git fetch --unshallow
>           old_sha=$(git rev-parse HEAD^1:GameServers/usvm)
>           new_sha=$(git rev-parse HEAD:GameServers/usvm)
>           if [ "$old_sha" != "$new_sha" ]; then
>             ./gradlew build
>           else
>             ./gradlew assemble
>           fi
> ```

Comment (emnigma) [reply .github/workflows/build_and_test_usvm.yaml]: > if it indeed does not work, we may try more direct approach:
> 
> ```yaml
>       - name: Check for submodule changes
>         run: |
>           git fetch --unshallow
>           old_sha=$(git rev-parse HEAD^1:GameServers/usvm)
>           new_sha=$(git rev-parse HEAD:GameServers/usvm)
>           if [ "$old_sha" != "$new_sha" ]; then
>             ./gradlew build
>           else
>             ./gradlew assemble
>           fi
> ```

This doesn't work completely: https://github.com/PySymGym/PySymGym/pull/103

CI on pull-request did launch `build` command

**Review by emnigma — APPROVED** [review]
Its ok as long as we do not change usvm version only

**Pull request merged**